Mathlib/Topology/Perfect.lean

```diff
@@ -215,3 +234,18 @@ theorem exists_perfect_nonempty_of_isClosed_of_not_countable [SecondCountableTop
 end Kernel
 
 end Basic
+
+section PerfectSpace
+
+variable {X : Type*} [TopologicalSpace X]
+
+instance PerfectSpace.not_isolated [PerfectSpace X] (x : X) : Filter.NeBot (𝓝[≠] x) := by
+  have := (PerfectSpace.univ_perfect X).acc _ (Set.mem_univ x)
+  rwa [AccPt, Filter.principal_univ, inf_top_eq] at this
+
+theorem perfectSpace_iff_forall_not_isolated : PerfectSpace X ↔ ∀ x : X, Filter.NeBot (𝓝[≠] x) :=
+  ⟨fun perfect x => perfect.not_isolated x, fun h_forall => ⟨⟨isClosed_univ, fun x _ => by
+    rw [AccPt, Filter.principal_univ, inf_top_eq]
+    exact h_forall x⟩⟩⟩
```

Comment:
If you add `@[mk_iff perfect_def]` above the definition of `Perfect` (you can see the generated lemma by hovering over its name), you can do:
```suggestion
theorem perfectSpace_iff_forall_not_isolated : 
    PerfectSpace X ↔ ∀ x : X, Filter.NeBot (𝓝[≠] x) := by
  simp [perfectSpace_def, perfect_def, Preperfect, AccPt]

instance PerfectSpace.not_isolated [PerfectSpace X] (x : X) : Filter.NeBot (𝓝[≠] x) := 
  perfectSpace_iff_forall_not_isolated.mp ‹_› x
```
Note that we don't need `preperfect_def` because `Preperfect` is a def so you can just rewrite through it.